central/complianceoperator/v2/benchmarks/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,195 @@
+//go:build sql_integration
+
+package datastore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	benchmarkStorage "github.com/stackrox/rox/central/complianceoperator/v2/benchmarks/store/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+const (
+	uuidStub1       = "933cf32f-d387-4787-8835-65857b5fdbfd"
+	uuidStub2       = "8f9850a8-b615-4a12-a3da-7b057bf3aeba"
+	uuidNonExisting = "1e52b778-63f2-4eab-aa81-c9b6381ceb02"
+)
+
+func TestComplianceBenchmarkDataStore(t *testing.T) {
+	suite.Run(t, new(complianceBenchmarkDataStoreSuite))
+}
+
+type complianceBenchmarkDataStoreSuite struct {
+	suite.Suite
+
+	hasReadCtx  context.Context
+	hasWriteCtx context.Context
+	noAccessCtx context.Context
+
+	mockCtrl *gomock.Controller
+
+	datastore DataStore
+	storage   benchmarkStorage.Store
+	db        *pgtest.TestPostgres
+}
+
+func (s *complianceBenchmarkDataStoreSuite) SetupSuite() {
+	s.T().Setenv(features.ComplianceEnhancements.EnvVar(), "true")
+	if !features.ComplianceEnhancements.Enabled() {
+		s.T().Skipf("Skip test when %s is disabled", features.ComplianceEnhancements.EnvVar())
+		s.T().SkipNow()
+	}
+}
+
+func (s *complianceBenchmarkDataStoreSuite) SetupTest() {
+	s.hasReadCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.ComplianceOperator)))
+	s.hasWriteCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.ComplianceOperator)))
+	s.noAccessCtx = sac.WithGlobalAccessScopeChecker(context.Background(), sac.DenyAllAccessScopeChecker())
+
+	s.mockCtrl = gomock.NewController(s.T())
+
+	s.db = pgtest.ForT(s.T())
+	s.storage = benchmarkStorage.New(s.db)
+	s.datastore = New(s.storage)
+}
+
+func (s *complianceBenchmarkDataStoreSuite) TearDownTest() {
+	s.db.Teardown(s.T())
+}
+
+func (s *complianceBenchmarkDataStoreSuite) TestUpsertBenchmark() {
+	// make sure we have nothing
+	benchmarkIDs, err := s.storage.GetIDs(s.hasReadCtx)
+	s.Require().NoError(err)
+	s.Require().Empty(benchmarkIDs)
+
+	b1 := getTestBenchmark(uuidStub1, "b1", "1.0", 1)
+	b2 := getTestBenchmark(uuidStub2, "b2", "2.0", 2)
+
+	s.Assert().NoError(s.datastore.UpsertBenchmark(s.hasWriteCtx, b1))
+	s.Assert().NoError(s.datastore.UpsertBenchmark(s.hasWriteCtx, b2))
+
+	count, err := s.storage.Count(s.hasReadCtx, search.EmptyQuery())
+	s.Assert().NoError(err)
```

Comment:
You can use `Require` every time following assertions would fail if this assertions already failed.
The following assertions in such a case only create noise.

```suggestion
	s.Require().NoError(err)
```

Reply:
Done